sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -190,3 +190,43 @@ size_t FileBodyStream::OnRead(uint8_t* buffer, size_t count, Azure::Core::Contex
 void FileBodyStream::Rewind() { m_randomAccessFileBodyStream->Rewind(); }
 
 int64_t FileBodyStream::Length() const { return m_randomAccessFileBodyStream->Length(); }
+
+ProgressBodyStream::ProgressBodyStream(
+    BodyStream* bodyStream,
+    std::function<void(int64_t bytesTransferred)> callback)
+    : m_bodyStream(bodyStream), m_bytesTransferred(0), m_callback(std::move(callback))
+{
+  if (bodyStream == nullptr)
+  {
+    throw std::runtime_error("Parameter 'bodystream' cannot be null");
+  }
```

Comment:
This should be AZURE_ASSERT. But it's even better if we don't use raw pointer and update to unique_ptr

Reply:
done assert